README.md

```diff
@@ -1,2 +1,89 @@
-# puppetlabs-powershell_task_helper
-A helper library for Puppet Tasks written in PowerShell
+
+# powershell_task_helper
+
+A PowerShell helper library for use by [Puppet Tasks](https://puppet.com/docs/bolt/1.x/writing_tasks.html). It provides a class that handles error generation, simplifies JSON input and output, and makes testing your task easier. It requires Bolt >= 1.1 and Puppet Enterprise >= 2019.0.
+
+#### Table of Contents
+
+1. [Description](#description)
+2. [Setup - The basics of getting started with powershell_task_helper](#setup)
+    * [Beginning with powershell_task_helper](#beginning-with-powershell_task_helper)
+3. [Usage - Configuration options and additional functionality](#usage)
+4. [Limitations - OS compatibility, etc.](#limitations)
+5. [Development - Guide for contributing to the module](#development)
+
+## Description
+
+This library handles parsing JSON input, serializing the result as JSON output, and producing a formatted error message for errors.
+
+## Setup
+
+### Beginning with powershell_task_helper
+
+To use this library, include this module in a `Puppetfile`
+```ruby
+mod 'puppetlabs-powershell_task_helper'
+```
+
+Add it to your [task metadata](https://puppet.com/docs/bolt/1.x/writing_tasks.html#concept-677)
+```json
+{
+  "files": ["powershell_task_helper/files/task_helper.ps1"],
+  "input_method": "powershell"
+}
+```
+
+## Usage
+
+When writing your task include the library in your script by dot-sourcing the `task_helepr.ps1` file
+
+```powershell
+#!/usr/bin/env pwsh
+[CmdletBinding()]
+Param(
+  [Parameter(Mandatory = $True)]
+  [String]
+  $Name
+)
+
+function Get-CurrentDirectory
+{
+  $thisName = $MyInvocation.MyCommand.Name
+  [IO.Path]::GetDirectoryName((Get-Content function:$thisName).File)
+}
+
+$taskHelper = Join-Path (Get-CurrentDirectory) '../../powershell_task_helper/files/task_helper.ps1'
+. $taskHelper
```

Comment:
Rather than dot sourcing, this would likely be better if Bolt could add a path for modules to automatically be found in `$PSModulePath` via `Import-Module`